src/plus/gk/checkin.ts

```diff
@@ -176,7 +180,18 @@ export function getSubscriptionFromCheckIn(
 		);
 	}
 
-	if (effective == null || getSubscriptionPlanPriority(actual.id) >= getSubscriptionPlanPriority(effective.id)) {
+	const isActualLicenseExpired =
+		computeSubscriptionState({
+			plan: {
+				actual: actual,
+				effective: actual,
+			},
+			account: account,
+		}) === SubscriptionState.PaidExpired;
+	if (
+		effective == null ||
+		(getSubscriptionPlanPriority(actual.id) >= getSubscriptionPlanPriority(effective.id) && !isActualLicenseExpired)
+	) {
```

Comment:
One other issue I am seeing with `computeSubscriptionState` that could happen if we let expired subscriptions through:

1. You have a paid, but expired, Pro license.
2. You are on a (reactivated) Pro trial.

Note that this conditional is not triggered so `effective = { ...actual }` does not happen, but...

`effective` is still set in 171 to a trial license with `effective.id === SubscriptionPlanId.Pro`. And `actual` is an expired license with `actual.id === SubscriptionPlanId.Pro`.

So when we hit `computeSubscriptionState`, it will match the `actual.id === `effective.id` check (which previously wouldn't happen unless we hit this `effective = { ...actual }` line, and that would only happen if it was an active paid license since we were filtering out paid licenses that were expired.

Ultimately, this would cause your license to show as a paid Pro license rather than an active Pro trial.

To avoid that regression, I suggest we just attach an `isTrial` property to these and we can add that to the condition in `computeSubscriptionState`.

Reply:
Updated with all feedback except this one